### PR TITLE
feat(sdk): Add more Sliding Sync logs

### PR DESCRIPTION
**Pull request by Hywan**
This patch adds more logs around `SlidingSync` to ease debugging sessions.

I'm not totally sure those logs will be helpful, but at least it will help to detect when a request and a response are sent as expected.

### CODE REVIEW

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 1]
```diff
@@ -1058,6 +1069,8 @@ impl SlidingSync {
         // coming from the `OlmMachine::outgoing_requests()` method.
         #[cfg(feature = "e2e-encryption")]
         let response = {
+            debug!("Sliding Sync is sending the request _with_ E2EE");
```
Comment (poljar) [inline crates/matrix-sdk/src/sliding_sync/mod.rs:1072]: Not sure if it's clear to you but this is sending the sliding sync request and all the outgoing E2EE related requests out, if any. From the log line it sounds like it's enabling the E2EE extension in the sliding sync request.

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 2]
```diff
@@ -1088,12 +1107,12 @@ impl SlidingSync {
         // Spawn a new future to ensure that the code inside this future cannot be
         // cancelled if this method is cancelled.
         spawn(async move {
-            debug!("Sliding sync response received");
+            debug!("Sliding Sync response handing starts");
 
             // In case the task running this future is detached, we must be
-            // ensure responses are handled one at a time, hence we lock the
+            // ensured responses are handled one at a time, hence we lock the
```
Comment (poljar) [inline crates/matrix-sdk/src/sliding_sync/mod.rs:1113]: Not sure that this sounds better to me. I think removing the "be" would improve it, i.e. "we must ensure" sounds correct to me.

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 3]
```diff
@@ -1119,13 +1138,14 @@ impl SlidingSync {
             // happens here.
             let sync_response = this.inner.client.process_sliding_sync(&response).await?;
 
-            debug!("Sliding sync response has been processed");
+            debug!(?sync_response, "Sliding Sync response has been processed");
```
Comment (poljar) [inline crates/matrix-sdk/src/sliding_sync/mod.rs:1141]: Perhaps consistent verbs here would clarify things further.
```suggestion
            debug!(?sync_response, "Sliding Sync response has been handled");
```

I don't care if we use "handled" or "processed", lets just stick to one. So if you prefer the other one feel free to change things up.

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 4]
```diff
@@ -1034,19 +1041,23 @@ impl SlidingSync {
...
+            // request. We use the (optional) `txn_id` field for that.
+            txn_id: Some(stream_id.to_owned()),
+            timeout: Some(timeout),
+            lists,
+            room_subscriptions,
+            unsubscribe_rooms,
+            extensions,
+        });
+
+        debug!(?request, "Sliding Sync will send this request");
```
[the remark this answers is not in the document]
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/mod.rs]: OK, removed.

File: crates/matrix-sdk/src/sliding_sync/mod.rs (hunk 1 above)
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/mod.rs]: I've understood it right, but the phrasing of the logs is bad. Rephrased.

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 5]
```diff
@@ -1088,12 +1107,12 @@ impl SlidingSync {
         // Spawn a new future to ensure that the code inside this future cannot be
         // cancelled if this method is cancelled.
         spawn(async move {
-            debug!("Sliding sync response received");
+            debug!("Sliding Sync response handing starts");
```
[the remark this answers is not in the document]
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/mod.rs]: Thanks!

File: crates/matrix-sdk/src/sliding_sync/mod.rs (hunk 2 above)
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/mod.rs]: Correct :-)!

File: crates/matrix-sdk/src/sliding_sync/mod.rs (hunk 3 above)
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/mod.rs]: Yeah, but “processed” here is different. I'll try to find a better phrasing entirely.

**Pull request merged**